hpx/components/containers/coarray/detail/are_integral.hpp

```diff
@@ -0,0 +1,31 @@
+//  Copyright (c) 2017 Antoine Tran Tan
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file hpx/components/coarray/detail/are_integral.hpp
+
+#ifndef HPX_ARE_INTEGRAL_HPP
+#define HPX_ARE_INTEGRAL_HPP
+
+#include <type_traits>
+
+///////////////////////////////////////////////////////////////////////////////
+/// \cond NOINTERNAL
+
+namespace hpx {  namespace detail {
+
+    template<bool ...>
+    struct bools;
+
+    template<typename ... T>
+    struct are_integral
+    : public std::integral_constant< bool,
+        std::is_same<
+            bools<true, std::is_integral<T>::value ...>,
+            bools<std::is_integral<T>::value ...,true > >::value >
+    {};
+
```

Comment:
Wouldn't it be simpler to just use `hpx::util::detail::all_of<std::is_integral<T>::value...>` instead (see https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/detail/pack.hpp#L62-L81)?